pkg/hubble/parser/threefour/parser.go

```diff
@@ -316,13 +316,20 @@ func (p *Parser) resolveEndpoint(ip net.IP, datapathSecurityIdentity uint32) *pb
 	if p.endpointGetter != nil {
 		if ep, ok := p.endpointGetter.GetEndpointInfo(ip); ok {
 			epIdentity := resolveIdentityConflict(ep.GetIdentity())
-			return &pb.Endpoint{
+			e := &pb.Endpoint{
 				ID:        uint32(ep.GetID()),
 				Identity:  epIdentity,
 				Namespace: ep.GetK8sNamespace(),
 				Labels:    sortAndFilterLabels(p.log, ep.GetLabels(), epIdentity),
 				PodName:   ep.GetK8sPodName(),
 			}
+			if pod := ep.GetPod(); pod != nil {
+				for _, owner := range pod.GetOwnerReferences() {
```

Comment:
could be nice to pre-allocate these based on the size of `GetOwnerReferences`

Reply:
Updated.